### ISSUE TITLE
Trusted keys and alt marketplaces

### ISSUE THREAD

**Issue by MattDHill** [Start9Labs/embassy-os#435]

**Comment by ProofOfKeags** [comment]
This seems like it might be two features:
1. Add the ability to add/replace package signing keys from the developer/maintainer
2. Add the ability to add/replace the package repository (registry)

**Comment by MattDHill** [comment]
Correct, these are separate features, but with multiple marketplaces, trusted keys will become an important protection for users.

Also, services will likely be identified by the conjunction of their ID and the key(s) used to sign the binary. Probably tangential, but an important point.

**Comment by ProofOfKeags** [comment]
Ok so I will separate them out into their own items. From the sound of your comment, the alt marketplaces one is contingent on having key management infrastructure so that ought to be worked on first.

**Comment by ProofOfKeags** [comment]
The design has been punted since the package format is versioned. How we handle trust chains will be dealt with in the next package version.

**Comment by ProofOfKeags** [comment]
0.3.1 SoW: Scope this completely. Do not need to implement for 0.3.1

**Comment by MattDHill** [comment]
Packages signed with developer pubkey

registries return list of trusted pubkeys for each service id with names

registries return list of pubkeys to revoke

sideloading packages with untrusted pubkey will prompt to trust pubkey

installing dependencies with untrusted pubkey will prompt to trust pubkey

page for managing trusted and revoked pubkeys

**Comment by MattDHill** [comment]
Will rescope as part of registry rework
